### ISSUE TITLE
[JIT] Allow Named Arguments with torch.jit._fork

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#38357]
## 🐛 Bug / Feature

Named Arguments work with `torch.jit_fork` in script mode, but not in eager / tracing.
```
foo(x, y: int):
    return x + y

def caller(x):
    return torch.jit._wait(torch.jit._fork(foo, y=2, x=x))
```

```
print(torch.jit.script(caller).graph)`
graph(%x.1 : Tensor):
  %1 : int = prim::Constant[value=2]() # test/test_jit.py:16712:58
  %20 : Future[Tensor] = prim::fork_0(%x.1, %1) # test/test_jit.py:16712:35
  %7 : Tensor = aten::wait(%20) # test/test_jit.py:16712:19
  return (%7)
with prim::fork_0 = graph(%x.1 : Tensor,
      %6 : int):
  %1 : Function = prim::Constant[name="foo"]()
  %4 : Tensor = prim::CallFunction(%1, %x.1, %6) # test/test_jit.py:16712:35
  return (%4)
```
`torch.jit.trace(caller, (torch.tensor(2),))`
> TypeError: fork(): incompatible function arguments. The following argument types are supported:
    1. (*args) -> torch._C.Future

Being able to used named arguments is nice for readability. 

cc @suo

**Comment by eellison** [comment]
The fix to this would involve adding kwargs as an argument to https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/python/init.cpp#L865. The tests for fork/join are in https://github.com/pytorch/pytorch/blob/master/test/jit/test_async.py.

**Comment by wconstab** [comment]
closed by 6fd48e2
